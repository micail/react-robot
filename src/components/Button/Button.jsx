import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = ({ name, func }) => {
  return (<button type="button" onClick={func} className="btn" id={`btn-${name}`}>{name.toUpperCase()}</button>);
};
export default Button;

Button.propTypes = {
  name: PropTypes.string,
  func: PropTypes.func,
};

Button.defaultProps = {
  name: '',
  func: null,
};
