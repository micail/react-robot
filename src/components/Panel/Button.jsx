import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = (props) => {
  const name = Object.keys(props).toString();
  return (<button type="button" className="btn" id={`btn-${name}`}>{name.toUpperCase()}</button>);
};
export default Button;

Button.propTypes = {
};

Button.defaultProps = {
};
