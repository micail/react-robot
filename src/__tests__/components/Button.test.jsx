import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../components/Button/Button';

describe('Button component', () => {
  let component;
  beforeAll(() => {
    component = shallow(<Button />);
  });

  it('should render button component', () => {
    expect(component).toMatchSnapshot();
  });
});
