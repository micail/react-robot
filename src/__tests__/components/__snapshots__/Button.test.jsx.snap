// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render button component 1`] = `
<button
  className="btn"
  id="btn-"
  type="button"
/>
`;
